### PR TITLE
force unbond in staking

### LINKED ISSUE #414: Staking `force_unbond`
Currently, as a nominator or validator, your request to fully unbond can fail if you are not chilled yet. 

The right fix is for the UI to detect this and batch it with a `chill`.

A `force_unbond` can help clarify this, and is not that much overhead on the chain runtime. under the hood, it would be an optional chill, followed by unbond.

### PR DESCRIPTION
**Pull request by dharjeezy**
closes #414 

Polkadot address: 12GyGD3QhT4i2JJpNzvMf96sxxBLWymz4RdGCxRH5Rj5agKW

### CODE REVIEW

File: substrate/frame/staking/src/pallet/mod.rs [hunk 1]
```diff
@@ -1978,6 +1906,32 @@ pub mod pallet {
...
+		]
+		pub fn force_unbond(
+			origin: OriginFor<T>,
+			#[pallet::compact] value: BalanceOf<T>,
+		) -> DispatchResultWithPostInfo {
+			let controller = ensure_signed(origin)?;
+
+			let ledger = Self::ledger(StakingAccount::Controller(controller.clone()))?;
+
+			Self::chill_stash(&ledger.stash);
```
Comment (Ank4n) [inline substrate/frame/staking/src/pallet/mod.rs:1924]: It should only chill if unbond will fail without chilling right? i.e. when unbonding would make the bonded balance go below `MinNominatorBond`.

Also, please add tests.

Comment (dharjeezy) [reply substrate/frame/staking/src/pallet/mod.rs]: I have added the check for full unbonding.

i have also included test

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5500245

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5500535

**Comment by Ank4n** [comment]
> Code looks good. Reflecting more on the name, I would call this "fully unbond", meaning it will chill and unbond everything. Also note that this is merely a shorthand for batching the aforementioned.

This extrinsic takes a value so not a full unbond. If the intention was only to do chill for full unbond, imo better to add this in the existing `unbond` check that does the chilling if it is a `full unbond`.

**Comment by dharjeezy** [comment]
> > Code looks good. Reflecting more on the name, I would call this "fully unbond", meaning it will chill and unbond everything. Also note that this is merely a shorthand for batching the aforementioned.
> 
> This extrinsic takes a value so it is not always a full unbond. If the intention was only to do chill for full unbond, imo better to add this in the existing `unbond` check that does the chilling if it is a `full unbond`.

@Ank4n according to @kianenigma we want to fully unbond everything after chilling. Therefore, the extrinsic don't need to take any value, also there will be no need to add a check for minimum active bond which you suggested earlier here. https://github.com/paritytech/polkadot-sdk/pull/3629#discussion_r1519427329

Also, if want to do a chill also when we expect a value, we can definitely do that in the unbond extrinsic. should i go ahead and include that?

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630107

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630125

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630396

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630397

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630676

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5630828

**Comment by dharjeezy** [comment]
Apologies please, Wrongfully messed up the commit here.
the new PR is https://github.com/paritytech/polkadot-sdk/pull/3811

**Pull request closed without merging**